### DISCUSSION TITLE
How do I align the text?

### DISCUSSION THREAD

**Discussion by himch** [ImageMagick/ImageMagick#2854] [Help]
Hi, please help me(.
![x1x000](https://user-images.githubusercontent.com/11749638/99907787-ea9e9780-2cef-11eb-914c-4ccbd3c8c7ef.png)

I place my caption below the photo, the text is centered and cropped on both sides when the caption line is long.
I want to align the text to the left so that only the right side of the line is cut.
How can I achieve this?

My code:
magick montage * .jpg -geometry 240x240! + 0 + 0 -set label "very very very long long string" -pointsize 20 -tile 1x1 x1x% 03d.png
![x1x000]

**Comment by fmw42** [comment]
It does not look like montage accepts -gravity west for control of the position of your text label. So I think what you ask is currently not possible. I suggest you use label: to create your text image of the width you want and then use magick to append it below your image. With label: the font size will adjust so that the text will fit completely. You can write a script to loop over all your images and append the same label image to all your images. Otherwise, create an image the size you want and use -annotate to add the text adjusted to start at the left.

What is your platform? Scripting is different in Windows vs Unix-like systems

Here is an example of the former in Unix syntax:

`magick rose: -resize 240x240! \( -size 240x30 -background white -gravity center label:"very very very long long string" \) -alpha off -append rose_text1.png`

![rose_text1](https://user-images.githubusercontent.com/13458218/99909739-24937d80-2c9f-11eb-8f71-75b089de4825.png)


Here is an example of the latter in Unix syntax:

`magick rose: -resize 240x240! \( -size 240x30 xc:white -pointsize 20 -gravity west -annotate +0+0 "very very very long long string" \) -alpha off -append rose_text2.png`

![rose_text](https://user-images.githubusercontent.com/13458218/99909657-b77fe800-2c9e-11eb-987e-7a1eb58856cc.png)


P.S. there should be no spaces in *.jpg or 240x240!+0+0. You command above has extraneous spaces which should keep it from working properly. Probably a typo.

**Reply by himch** [reply]
Thank you, Fred.

I am writing in Python in PyCharm on Windows, the code will work on Ubuntu. I think I'll use Python to cut the lines so that they fit under the picture.

**Comment by fmw42** [comment]
You can use Python Wand, which uses ImageMagick if you want. I am not familiar with PyCharm. Is that pure Python? Does it use ImageMagick?
